### ISSUE TITLE
Save original state of receipt

### ISSUE THREAD

**Issue by taylordowns2000** [OpenFn/core#74]
When a receipt is edited the original state of the receipt should be saved and a "revert" button should be added that sets it back to the original.

**Comment by taylordowns2000** [comment]
1. BEFORE receipt is updated...
2. If `backup_body` is blank...
3. set `backup_body = body` - this needs to be the body from the DB, not from the form changeset... so it should happen in the controller?
4. never do that if `backup_body` is NOT blank.
5. add "Restore Original Body" button to receipt edit form which takes value from `backup_body` and fills in the `body` input field on the redux-form.

**Comment by stuartc** [comment]
@taylordowns2000 ouch, this is harder to think about than I thought..

I'd really like to avoid having 2 copies of the same data, (in the revert case).
Also it feels super complicated (or hacky at least) to have a switcher constantly making the decision about what/how to update.

**Comment by taylordowns2000** [comment]
written well, this code usually never fires. and if it does fire, it only fires once... duplicating the original data before the update takes place. i don't know about the implementation, but in theory that's just what we're trying to do, no?

**Comment by stuartc** [comment]
Yeah my concern is around implementation.. Let me think about the db side of things - I'd like to avoid having body not always being the source of truth - at least from the model side of things..
